test/ServiceDiscoveryTest.cs

```diff
@@ -192,42 +187,46 @@ public void Discover_ServiceInstance()
         [TestMethod]
         public void Discover_ServiceInstance_WithAnswersContainingAdditionRecords()
         {
-            var service = new ServiceProfile("y", "_sdtest-2._udp", 1024);
+            var service = new ServiceProfile("y", "_sdtest-2._udp", 1024, new[] { IPAddress.Parse("127.1.1.1") });
             var done = new ManualResetEvent(false);
-            var mdns = new MulticastService();
-            var sd = new ServiceDiscovery(mdns)
-            {
-                AnswersContainsAdditionalRecords = true
-            };
-            Message discovered = null;
-            mdns.NetworkInterfaceDiscovered += (s, e) =>
-            {
-                sd.QueryServiceInstances(service.ServiceName);
-            };
 
-            sd.ServiceInstanceDiscovered += (s, e) =>
+            using (var mdns = new MulticastService())
+            using (var sd = new ServiceDiscovery(mdns) { AnswersContainsAdditionalRecords = true })
             {
-                if (e.ServiceInstanceName == service.FullyQualifiedName)
+                Message discovered = null;
+
+                mdns.NetworkInterfaceDiscovered += (s, e) =>
                 {
-                    Assert.IsNotNull(e.Message);
-                    discovered = e.Message;
-                    done.Set();
-                }
-            };
-            try
-            {
+                    sd.QueryServiceInstances(service.ServiceName);
+                };
+
+                sd.ServiceInstanceDiscovered += (s, e) =>
+                {
+                    if (e.ServiceInstanceName == service.FullyQualifiedName)
+                    {
+                        Assert.IsNotNull(e.Message);
+                        discovered = e.Message;
+                        done.Set();
+                    }
+                };
+
                 sd.Advertise(service);
+
                 mdns.Start();
+
                 Assert.IsTrue(done.WaitOne(TimeSpan.FromSeconds(1)), "instance not found");
-                Assert.AreEqual(0, discovered.AdditionalRecords.Count);
-                Assert.IsTrue(discovered.Answers.Count > 1);
-            }
-            finally
-            {
-                sd.Dispose();
-                mdns.Stop();
+
+                var additionalRecordsCount =
+                    1 + // SRVRecord
+                    1 + // TXTRecord
+                    1; // AddressRecord
+
+                var answersCount = additionalRecordsCount +
+                    1; // PTRRecord
+
+                Assert.AreEqual(additionalRecordsCount, discovered.AdditionalRecords.Count);
+                Assert.AreEqual(additionalRecordsCount + 1, discovered.Answers.Count);
```

Comment:
Why all the changes the to an existing test?

Reply:
It looks for me, prev test was written to cover invalid "AnswersContainsAdditionalRecords" functionality.
As this "AnswersContainsAdditionalRecords" was a bit fixed. The test doesn't work anymore.
Please check this on master.